src/components/Select/Select.component.tsx

```diff
@@ -81,23 +85,47 @@ export const Select = (props: SelectProps) => {
 		setIsOpen(true);
 	};
 
-    /**
-     * On clicking the trailing icon the dropdown is opened or closed with respect to the state of the select
-     *
-     * if isOpen = `true`  a close icon is rendered, on clicking clears the input field and closes the dropdown
-     * if isOpen = `false` an arrow down is rendered, on clicking opens the dropdown
-     *
-     */
-    const onIconTrailingIconClickHandler = () => {
-        setValue('');
-        if (isOpen) {
-            inputRef.current.blur();
-            setIsOpen(false);
-        } else {
-            inputRef.current.focus();
-            setIsOpen(true);
-        }
-    };
+	/**
+	 * On clicking the trailing icon the dropdown is opened or closed with respect to the state of the select
+	 *
+	 * if isOpen = `true`  a close icon is rendered, on clicking clears the input field and closes the dropdown
+	 * if isOpen = `false` an arrow down is rendered, on clicking opens the dropdown
+	 *
+	 */
+	const onIconTrailingIconClickHandler = () => {
+		setValue('');
+		setActive(0);
+		if (isOpen) {
+			inputRef.current.blur();
+			setIsOpen(false);
+		} else {
+			inputRef.current.focus();
+			setIsOpen(true);
+		}
+	};
+
+	/**
+	 * The function is called when an event is detected on the keyboard,
+	 * so you can browse through the list and select one.
+	 */
+	const onKeyDown: React.KeyboardEventHandler<HTMLInputElement> = (event) => {
+		if (event.key === 'Enter') {
+			const option = options[active];
+			setValue(option.label ?? option.value);
+			onSelectOptionHandler(event, option);
```

Comment:
don't we need to set `setActive(0)` here?

Reply:
I think no, because `on enter`, the active item will be the selected one